### PR TITLE
Add dbus/1.12.16

### PR DESCRIPTION
**Pull request by SemyonDuB**
Specify library name and version:  **dbus/1.12.16**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:66]: Are the names with conan's cmake files wrong? Or does dbus use its own cmake `find_XXX` scripts?

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:83]: Does not matter much, but looks nicer.
```suggestion
            self._cmake.definitions["DBUS_BUILD_TESTS"] = False
            self._cmake.definitions["DBUS_ENABLE_DOXYGEN_DOCS"] = False
            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = False
```

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:100]: This will call the cmake script directly.
You're also not using the `cmake` generator, whose `conan_basic_setup()` function will set the compiler flags correctly.

This is required especially for  MSVC. Because the `cmake` build helpers will configure the runtime `M[T]d?`

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:112]: Please remove the cmake scripts.
And also eventual `.pc` files.

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:115]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "DBus1"
        self.cpp_info.names["cmake_find_package_multi"] = "DBus1"
        self.cpp_info.libs = tools.collect_libs(self)
```
Don't forget to set the name for the `pkg_config` generator, if that would be different from `dbus.pc`.

File: recipes/dbus/all/test_package/CMakeLists.txt
Comment (madebr) [inline recipes/dbus/all/test_package/CMakeLists.txt:12]: Doesn't/Shouldn't Dbus1 create a `dbus1::dbus1` target (or something similar)?
That would avoid the `include_directories` call, and you can link directly to `dbus1::dbus1`.

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:23]: gcov = code coverage
I don't think this option is useful.

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:22]: Does this increase dbus' verbosity? Or only the verbosity of the build process?

File: recipes/dbus/all/conanfile.py
Comment (madebr) [inline recipes/dbus/all/conanfile.py:18]: Double negation (`disable_assert=False`) does not make readable code.
I would change these to `enable_assert` and `enable_checks`.

File: recipes/dbus/all/conanfile.py
Comment (SemyonDuB) [reply recipes/dbus/all/conanfile.py:66]: D-Bus uses own FindGLib2.cmake, but I also think that I've just mistake with X11, because I used libX11 from the bincrafters and forget about remove it from 64-66 lines.

File: recipes/dbus/all/conanfile.py
Comment (SemyonDuB) [reply recipes/dbus/all/conanfile.py:22]: I just added all of this lines from CMakeCache and ccmake, so I don't really know what they mean.

File: recipes/dbus/all/test_package/CMakeLists.txt
Comment (SemyonDuB) [reply recipes/dbus/all/test_package/CMakeLists.txt:12]: I didn't find anything about dbus1::dbus1, so I've just added this example from stackoverflow (https://stackoverflow.com/questions/38515316/how-to-configure-dbus-dependencies-with-cmake)

Comment (madebr) [reply recipes/dbus/all/test_package/CMakeLists.txt:12]: I just checked the dbus1 of Fedora Linux (my distro).
See https://fedora.pkgs.org/31/fedora-x86_64/dbus-devel-1.12.16-3.fc31.x86_64.rpm.html
It distributes `DBus1Config.cmake` and `dbus-1.pc`, so the cmake name should be `DBus1`, the pkg_config name `dbus-1`.
Inside the cmake script, it only creates a `dbus-1` library.

Comment (madebr) [reply recipes/dbus/all/test_package/CMakeLists.txt:12]: So you should be able to do:
```
find_package(DBus1 REQUIRED)

add_executable(example example.cpp)
target_link_libraries(example dbus-1)
```

File: recipes/dbus/all/conanfile.py
Comment (Croydon) [reply recipes/dbus/all/conanfile.py:22]: Please make sure to only add options which are relevant for dbus consumers

**Comment by SSE4** [comment]
```

----Running------
> bin/example
-----------------
Connection Error:
Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead


```
I think we don't need to check that it can actually connect somewhere, we only need to check it compiles and links successfully.

**Comment by uilianries** [comment]
As @SSE4 commented, you don't need to run an usable example, you could call a method which shows the version, or call a helper which doesn't require X11 for instance. The idea for test package is checking the generated package only, if all artifacts are there, by building and linking the project. If you use an real example, it's a bonus.

**Comment by ericLemanissier** [comment]
@SemyonDuB This package is required for https://github.com/conan-io/conan-center-index/pull/1986. Do you still have some time to finish it ?

File: recipes/dbus/1.x.x/conanfile.py
Comment (Croydon) [inline recipes/dbus/1.x.x/conanfile.py:40]:
```suggestion
    def configure(self):
```

Raise `ConanInvalidConfiguration` in `configure()`

File: recipes/dbus/1.x.x/conanfile.py
Comment (Croydon) [inline recipes/dbus/1.x.x/conanfile.py:7]:
```suggestion
    license = ("AFL-2.1", "GPL-2.0-or-later")
```

https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12/COPYING#L1

File: recipes/dbus/1.x.x/conanfile.py
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/conanfile.py:37]:
```suggestion
        "expat/2.2.10"
```

File: recipes/dbus/1.x.x/conanfile.py
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/conanfile.py:53]:
```suggestion
            self.requires("glib/2.67.0")
```

File: recipes/dbus/1.x.x/conanfile.py
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/conanfile.py:95]: why removing the `bin` folder ? it is allowed in CCI packages
```suggestion
        for i in ["var", "share", "etc"]:
```

File: recipes/dbus/1.x.x/test_package/CMakeLists.txt
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/test_package/CMakeLists.txt:9]:
```suggestion
add_executable(example example.c)
```
Can it be consumed from C projects ?

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

File: recipes/dbus/1.x.x/conanfile.py
Comment (SemyonDuB) [reply recipes/dbus/1.x.x/conanfile.py:95]: D-Bus generate `dbus-daemon`, `dbus-launch` and other execuable files in `bin` folder, but for some reasons executables linking at the system paths like `/lib/x86_64-linux-gnu/libdbus-1.so.3`, maybe it's dbus bug or I can how to change this behavior in conan or cmake, but I don't know how?!

Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:95]: isn't this just a problem of `LD_LIBRARY_PATH`?

**Comment by uilianries** [comment]
@SemyonDuB Your other user "Semyon Dubskikh" didn't accept the CLA yet. Could you please accept the CLA. Regards.

**Comment by ericLemanissier** [comment]
@SemyonDuB can you please take a look at https://github.com/SemyonDuB/conan-center-index/pull/1 ?

File: recipes/dbus/1.x.x/test_package/example.c
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/test_package/example.c:12]:
```suggestion
    return 0;
}
```
There was some undefined behaviour herer !

File: recipes/dbus/1.x.x/test_package/conanfile.py
Comment (ericLemanissier) [inline recipes/dbus/1.x.x/test_package/conanfile.py:17]:
```suggestion
            self.run(os.path.join("bin", "example"), run_environment=True)
```

File: recipes/dbus/1.x.x/test_package/example.c
Comment (SemyonDuB) [reply recipes/dbus/1.x.x/test_package/example.c:12]: Oh, right!

**Review by ericLemanissier — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@Croydon @madebr do you think this one is ok ?

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:42]: I don't think dbus is supported on iOS, tvOS or Android.
```suggestion
        if self.settings.os not in ("Linux", "FreeBSD"):
            raise ConanInvalidConfiguration("dbus is only supported on Linux")
```

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:21]: Are these options needed?
These look like development options.
I would not make these options available.
Also, disable assertions and checks by default.

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:72]: What does the `DBUS_INSTALL_SYSTEM_LIBS` option do?
What is installed?
Shouldn't these system libs get installed always?

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:98]: Instead of moving, you can set `CMAKE_INSTALL_DATAROOTDIR` to  `os.path.join(self.package_folder, "res", "shared"` in build.
See cmake's GNUInstallDirs for more variables.

Also, are the files in var, share and etc needed for development/running?
Can dbus find them at runtime?
If dbus does not know where to find them, then they are useless.
Maybe some environment variable needs to be set?

File: recipes/dbus/1.x.x/test_package/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake_find_package", "cmake"
```

File: recipes/dbus/1.x.x/test_package/CMakeLists.txt
Comment (madebr) [inline recipes/dbus/1.x.x/test_package/CMakeLists.txt:10]: Shouldn't you use the imported target by the `find_package(DBus1 REQUIRED)` above here?

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:109]:
```suggestion
        self.cpp_info.includedirs.extend([
            "include/dbus-1.0", "lib/dbus-1.0/include"])
```

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:110]:
```suggestion
        self.cpp_info.libs = ["dbus-1"]
```

**Review by SSE4 — APPROVED** [review]
I am approving, as it already took long, and I don't want to lose existing approvals, and requested changes are not that critical. I'll create a follow up PR from @madebr suggestion's

**Comment by ericLemanissier** [comment]
It looks like something is blocking auto merge

**Comment by ericLemanissier** [comment]
It looks like something is blocking auto merge, probably the fact that @uilianries requested some changes on an old commit (I think the same situation happened on another PR this week)

**Comment by ericLemanissier** [comment]
friendly ping @uilianries if you are ok with this PR, can you approve it, so that it can be merged ?

**Comment by prince-chrismc** [comment]
> It looks like something is blocking auto merge, probably the fact that @uilianries requested some changes on an old commit (I think the same situation happened on another PR this week)

This was supposed to be changed, might be a 🐛 bug in the CCI bot 😱

File: recipes/dbus/1.x.x/conanfile.py
Comment (prince-chrismc) [inline recipes/dbus/1.x.x/conanfile.py:105]: I can not find any official cmake modules, where does the name come from?

File: recipes/dbus/1.x.x/test_package/CMakeLists.txt
Comment (prince-chrismc) [inline recipes/dbus/1.x.x/test_package/CMakeLists.txt:7]: I think it might be work testing it with the pkg_config method https://stackoverflow.com/a/38520201/8480874

File: recipes/dbus/1.x.x/conanfile.py
Comment (prince-chrismc) [inline recipes/dbus/1.x.x/conanfile.py:32]: cmake_paths is a little unusual, can you please confirm its required, there was nothing obvious in upstream

File: recipes/dbus/1.x.x/conanfile.py
Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:105]: I guess it's from upstream https://gitlab.freedesktop.org/dbus/dbus/-/blob/master/cmake/DBus1Config.cmake.in

Comment (prince-chrismc) [reply recipes/dbus/1.x.x/conanfile.py:105]: That should be the following

```suggestion
        # FIXME: There should not be a namespace
        self.cpp_info.filenames["cmake_find_package"] = "DBus1"
        self.cpp_info.filenames["cmake_find_package_multi"] = "DBus1"
        self.cpp_info.names["cmake_find_package"] = "dbus-1"
        self.cpp_info.names["cmake_find_package_multi"] = "dbus-1"
```

File: recipes/dbus/1.x.x/conanfile.py
Comment (uilianries) [reply recipes/dbus/1.x.x/conanfile.py:21]: By default it's enabled: 

```
option (DBUS_DISABLE_ASSERT "Disable assertion checking" OFF)
...
if (UNIX AND NOT DBUS_DISABLE_ASSERT)
	# required for backtrace
	SET(CMAKE_C_FLAGS_DEBUG   "${CMAKE_C_FLAGS_DEBUG}   -Wl,--export-dynamic")
	SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Wl,--export-dynamic")
	add_definitions(-DDBUS_BUILT_R_DYNAMIC)
endif (UNIX AND NOT DBUS_DISABLE_ASSERT)
```
I also prefer it disabled, but it doesn't harm.

**Review by uilianries — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please, follow @madebr suggestions

File: recipes/dbus/1.x.x/conanfile.py
Comment (SemyonDuB) [reply recipes/dbus/1.x.x/conanfile.py:98]: D-Bus can read values from `CMAKE_INSTALL_<dir>`, but `cmake.install()` installs all definitions in the default path (`<package_folder>/share`, `<package_folder>/etc` ...) instead of `<package_folder>/res/share`, etc. I tried to specify definitions in my _configure_cmake method, in build and package (before `cmake.install()` ) methods separately, but seems I don't know how it work exactly.
Also, I notice that in conan documentation (https://docs.conan.io/en/latest/reference/build_helpers/cmake.html?highlight=cmake_install_datarootdir) datarootdir sets to the default folder, but in conan hooks datarootdir, sysconfdir, localstatedir with default paths is wrong. This is very strange behaviour in my opinion, why not just set this in the res folder by default in the conan itself?
So, I can specify GNU install dirs in my _configure_cmake like this:
```
self._cmake.definitions["CMAKE_INSTALL_SYSCONFDIR"] =  os.path.join(self.package_folder, "res", "etc")
self._cmake.definitions["CMAKE_INSTALL_LOCALSTATEDIR"] =  os.path.join(self.package_folder, "res", "var")
self._cmake.definitions["CMAKE_INSTALL_DATAROOTDIR"] =  os.path.join(self.package_folder, "res", "share")
```
And dbus understand all of this and print:
```
install datadir: /home/semyon/.conan/data/dbus/1.12.20/_/_/package/d10203385f1b609816fdb837a88c3e1c4e81f9de1c4e81f9f/res/share
```
...
But anyway instalation sets definition paths to the default (`-- Installing: /home/semyon/.conan/data/dbus/1.12.20/_/_/package/d10203385f1b609816fdb837a88c3de1c4e81f9f/share/dbus-1/system.d`), therefore anything installs in `pakcage_folder` and I should move them in `res`. I'll be very glad if you can help me with this situation and descripe what I'm doing wrong?

**Comment by ericLemanissier** [comment]
can we merge this PR, it does not seem to make any progress ? I'll do another PR to address all the remarks on this one directly after.

**Comment by madebr** [comment]
The options really need to be culled, or at least have their defaults re-evaluated.
Are we okay with removing options after merge?

**Comment by ericLemanissier** [comment]
@SemyonDuB You don't seem to have much time for this PR. Would you be ok if I did the changes  on top of your branch, in a branch on bincrafters'repo, and submit a new PR ? I'm pushing because this package transitively blocks [gtk](https://github.com/conan-io/conan-center-index/pull/3624)

**Comment by SemyonDuB** [comment]
@ericLemanissier, I would be very appreciative if you do this!

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [reply recipes/dbus/1.x.x/conanfile.py:21]: It's perhaps safer. It's always nice to have extra checks for IPC.

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [reply recipes/dbus/1.x.x/conanfile.py:72]: I checked:
this installs the libexpat libraries + runtime libraries.
So this option should always be false on cci.

See https://cmake.org/cmake/help/latest/module/InstallRequiredSystemLibraries.html

File: recipes/dbus/1.x.x/conanfile.py
Comment (madebr) [reply recipes/dbus/1.x.x/conanfile.py:98]: I checked the cmake script of dbus and it looks like the `share` folder is hardcoded.
So you're using the correct approach.

File: recipes/dbus/1.x.x/conandata.yml
Comment (madebr) [inline recipes/dbus/1.x.x/conandata.yml:2]: `1.12.20` is from July 2020.
`1.13.18` is available too.
Would that be a valid version too?

**Comment by prince-chrismc** [comment]
Since there is a replacement PR for this can we close mark as draft or close pretty please? 🙏

**Comment by madebr** [comment]
@uilianries 
It looks like you're blocking this pr.
Can you re-review please?

**Comment by Croydon** [comment]
I think it is better to review https://github.com/conan-io/conan-center-index/pull/3950

(I can't review anything else than version bumps right now reasonable)

**Comment by prince-chrismc** [comment]
Again please close this PR! It's been replaced!

**Comment by uilianries** [comment]
@SemyonDuB please, take a look on @madebr 's comments. The package version, cmake_paths, include_dirs I would say are important points to be considered. Regards!

**Comment by ericLemanissier** [comment]
I don't understand. @SemyonDuB agreed that I make another PR based on his work, which I did in #3950. Why do you keep reviewing this one ?

**Review by uilianries — APPROVED** [review]

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
superseded by #3950

**Comment by uilianries** [comment]
@ericLemanissier Sorry, to many PRs, closing this one.

**Pull request closed without merging**
